### ISSUE TITLE
filehandles leak

### ISSUE THREAD

**Issue by valenpo** [orientechnologies/orientdb#8950]
### Java Version: java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)
## Expected behavior  
No leak of filehandles 

## Actual behavior  
filehandles are leaking from orientdb  

## Steps to reproduce  
Add many databases with such pattern plocal:/path/to/db/random/$db_name
and start database.

some of file leaks stacktrace 
```
#3 /var/opt/mailarchiva/portman/database/archiva/portman/orole.cpm by thread:service-4 on Tue Jul 09 08:29:26 UTC 2019
        at java.io.RandomAccessFile.<init>(RandomAccessFile.java:244)
        at com.orientechnologies.orient.core.storage.fs.OFileClassic.openChannel(OFileClassic.java:805)
        at com.orientechnologies.orient.core.storage.fs.OFileClassic.open(OFileClassic.java:625)
        at com.orientechnologies.orient.core.storage.cache.local.OWOWCache.readNameIdMapV1(OWOWCache.java:1962)
        at com.orientechnologies.orient.core.storage.cache.local.OWOWCache.initNameIdMapping(OWOWCache.java:1766)
        at com.orientechnologies.orient.core.storage.cache.local.OWOWCache.loadRegisteredFiles(OWOWCache.java:479)
        at com.orientechnologies.orient.core.storage.disk.OLocalPaginatedStorage.initWalAndDiskCache(OLocalPaginatedStorage.java:632)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.open(OAbstractPaginatedStorage.java:371)
        at com.orientechnologies.orient.core.db.OrientDBEmbedded.open(OrientDBEmbedded.java:281)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.open(ODatabaseDocumentTx.java:903)
        at com.orientechnologies.orient.core.db.OPartitionedDatabasePool$DatabaseDocumentTxPooled.internalOpen(OPartitionedDatabasePool.java:440)
        at com.orientechnologies.orient.core.db.OPartitionedDatabasePool.openDatabase(OPartitionedDatabasePool.java:300)
        at com.orientechnologies.orient.core.db.OPartitionedDatabasePool.acquire(OPartitionedDatabasePool.java:261)
        at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.<init>(OrientBaseGraph.java:172)
        at com.tinkerpop.blueprints.impls.orient.OrientTransactionalGraph.<init>(OrientTransactionalGraph.java:78)
        at com.tinkerpop.blueprints.impls.orient.OrientGraph.<init>(OrientGraph.java:137)
        at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory$1.getGraph(OrientGraphFactory.java:87)
        at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory.getTx(OrientGraphFactory.java:224)
        at com.stimulus.archiva.hp.c(MailArchiva:116)
        at com.stimulus.archiva.hp.a(MailArchiva:168)
        at com.stimulus.archiva.hn.a(MailArchiva:89)
        at com.stimulus.archiva.database.DatabaseService.c(MailArchiva:188)
        at com.stimulus.archiva.domain.bv$a.a(MailArchiva:142)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

[fileleaks.tar.gz](https://yadi.sk/d/Lnnp3R3EuxaeCg)

**Comment by laa** [comment]
Hi @valenpo why do you think there is file leak? Not sure that I understand.

**Comment by laa** [comment]
Oh I got it. I will handle it a bit later.

**Comment by valenpo** [comment]
Hi @laa. We have limit to 65k file handles. After upgrade to v3 we run out of file handles on same data. 
So after downgrade to latest 2x version, all became normal and system didn't crash. 

On fileleaks.tar.gz I provide full file handles for JVM. After 6 min of work, system crash with no file handles.

thanks.

**Comment by laa** [comment]
@valenpo could you send me the start of the server log in 3.0.x

**Comment by laa** [comment]
@valenpo for the future Yandex is prohibited in my country, so better to use other file sharing service.

**Comment by valenpo** [comment]
> @valenpo for the future Yandex is prohibited in my country, so better to use other file sharing service.

Ok. Many things prohibited on my country too. I use vpn ;)

**Comment by laa** [comment]
@valenpo I have got the reason, I will fix it today or tomorrow.

**Comment by laa** [comment]
Fixed. Will release next week.

**Comment by valenpo** [comment]
@laa thanks!

**Comment by jamieb22** [comment]
Andrey. Unfortunately, this is bug is not fixed in Orient DB 3.0.24. When multiple Orient DB databases are opened in the multitenant scenario, the error we get is "too many open file handles". Please download a full stack of all orient db file handles at the exact time open file handles error occurred:  [orientdb_openfilehandle_logs.zip](https://ln2.sync.com/dl/d95d8c810/in5yybjh-qyk8vyrx-2idxk9g3-wmpambk2)
